internal/codegen/templates/delegate.tmpl

```diff
@@ -158,3 +171,48 @@ func (m *{{.Name | toLower}}CallbacksMap) delete(p unsafe.Pointer) {
 
 	delete(m.callbacks, p)
 }
+
+// typedEventHandlerReleaseChannelsMap keeps a map with channels
+// used to keep a goroutine alive during the lifecycle of this object.
+// This is required to avoid causing a deadlock error.
+// See this: https://github.com/golang/go/issues/55015
+type {{.Name | toLower}}ReleaseChannelsMap struct {
+	mu        *sync.Mutex
+	chans map[unsafe.Pointer]chan struct{}
+}
+
+func (m *{{.Name | toLower}}ReleaseChannelsMap) create(p unsafe.Pointer) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	c := make(chan struct{})
+	m.chans[p] = c
+
+	go func() {
+		// we need a timer to trick the go runtime into
+		// thinking there's still something going on here
+		// but we are only really interested in <-c
+		t := time.NewTimer(time.Minute)
+		for {
+			select {
+			case <-t.C:
+				t.Reset(time.Minute)
+			case <-c:
+				if !t.Stop(){
+					<-t.C
```

Comment:
I just followed what the docs say, but it's true that in this case nobody is going to be reading from the channel so it's safe to just stop.

```
Stop does not close the channel, to prevent a read from the channel succeeding incorrectly.

To ensure the channel is empty after a call to Stop, check the return value and drain the channel.
For example, assuming the program has not received from t.C already:

if !t.Stop() {
    <-t.C
}
```